Tests/SentryTests/Transaction/SentrySpanTests.swift

```diff
@@ -63,6 +63,131 @@ class SentrySpanTests: XCTestCase {
         clearTestState()
     }
     
+    func testSpanDoesNotIncludeTraceProfilerID() throws {
+        fixture.options.profilesSampleRate = 1
+        let span = fixture.getSut()
+        XCTAssert(SentryTraceProfiler.isCurrentlyProfiling())
+        span.finish()
+        
+        let serialized = span.serialize()
+        XCTAssertNil(serialized["profile_ids"])
+    }
+    
+    // test that a span that starts before a continuous profile and ends before
+    // the profile ends includes that profile id in the span
+    func testSpanWithProfileId() throws {
+        let span = fixture.getSut()
+        SentryContinuousProfiler.start()
+        let profileId = try XCTUnwrap(SentryContinuousProfiler.profiler()?.profilerId.sentryIdString)
+        span.finish()
+        
+        let serialized = span.serialize()
+        
+        let includedProfileIDs = try XCTUnwrap(serialized["profile_ids"] as? [String])
+        XCTAssert(includedProfileIDs.contains(profileId))
```

Comment:
`l`: I think we could assert that it only contains this one profileId and nothing else. That's what I would expect.

Reply:
This will change to exactly that when we make the change I mentioned: https://github.com/getsentry/sentry-cocoa/pull/4009#issuecomment-2136240595